packages/core/src/modules/vc/repository/W3cCredentialRecord.ts

```diff
@@ -44,12 +44,16 @@ export class W3cCredentialRecord extends BaseRecord<DefaultW3cCredentialTags, Cu
   }
 
   public getTags() {
+    // Contexts are usually strings, but can sometimes be objects. We're unable to use objects as tags,
+    // so we filter out the objects before setting the tags.
+    const stringContexts = this.credential.contexts.filter((ctx) => typeof ctx === 'string') as string[]
```

Comment:
I thought about that as well. However, let me explain my rationale.

The reason I've put this logic in `getTags()`, is because a `@context` can have many different forms. In this case, I was dealing with a credential with the following context:
```
'@context': [
      'https://www.w3.org/2018/credentials/v1',
      {
        '@vocab': 'https://w3id.org/security/undefinedTerm#'
      },
      'https://mattr.global/contexts/vc-extensions/v1',
      'https://purl.imsglobal.org/spec/ob/v3p0/context.json',
      'https://w3c-ccg.github.io/vc-status-rl-2020/contexts/vc-revocation-list-2020/v1.jsonld'
    ],
```
The `@vocab` property here, sets the default vocabulary. In other words, any type that isn't defined by one of the other contexts will fall back to the `@vocab` value. In this case, though, the `@vocab` value is also a URL. However, if you try to resolve the URL, you'll find that it's non-existent.

Apart from that, if you look at [example 7](https://www.w3.org/TR/json-ld11/#example-7-in-line-context-definition) of the JSON-LD spec, contexts that are objects can contain nested URLs as well.

Personally, I'd expect a getter called `contextUris` or `contextUrls` to return all the URLs inside the context (so also the ones that are deeply nested). Additionally, I'd expect all those URIs to be resolvable (assuming the corresponding services/domains are up).

My idea of putting this logic inside `getTags()` is that we filter out anything that can't be used as a tag only. Putting this logic in a dedicated getter on the `credential` class would make it specific to this use case, potentially causing confusion for other callers of this getter down the road.

That all being said, I'm happy to change it if you think I'm overthinking here ;)

Reply:
Ok makes sense I think. What I mostly want to prevent is that we get multiple implementations of this that do it slightly different. But your reasoning makes sense.